### PR TITLE
Anomaly Cores

### PR DESCRIPTION
**Pull request by TheShuEd**
image

## About the PR

This PR adds anomaly cores to the game. When an anomaly is detonated by a supercrit, a decaying anomaly core falls from it, which costs 10000 spesos. Over the course of 10 minutes, the cost of the core gradually drops to 200 spesos, and it becomes an inert core.

If the anomaly is destroyed with APE, the inert core will fall out of it.

## Why / Balance
[link](https://github.com/space-wizards/docs/blob/a3a9dd5a9ad5227e4da0300080afcb9195086ab8/src/en/proposals/anomaly-cores.md)

is the first part of implementing the new anomaly mechanic.

## Media

https://github.com/space-wizards/space-station-14/assets/96445749/150e9483-cea1-436d-be34-186136559b98

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: When the anomaly explodes, a very expensive core falls from it. Be sure to sell it in 10 minutes before it becomes useless.

### CODE REVIEW

**Comment by TheShuEd** [comment]
lol, collect the infinity gauntlet

**Comment by mirrorcult** [comment]
oh these sprites are really good, i hadnt considered a design like that

**Comment by UbaserB** [comment]
oh my god this looks so cool.

File: Content.Server/Anomaly/Effects/AnomalyCoreSystem.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            component.AccumulatedFrametime += frameTime;
+            if (component.AccumulatedFrametime < component.UpdateInterval)
+                continue;
+
+            var timeLeft = component.DecayMoment - _gameTiming.CurTime;
+            var lerp = (double)(timeLeft.TotalSeconds / component.TimeToDecay);
+            lerp = Math.Clamp(lerp, 0, 1);
+
+            if (TryComp<StaticPriceComponent>(uid, out var price))
+                price.Price = MathHelper.Lerp(component.FuturePrice, component.OldPrice, lerp);
```
Comment (mirrorcult) [inline Content.Server/Anomaly/Effects/AnomalyCoreSystem.cs:52]: this should absolutely not work by continually updating static price, just remove staticprice on the cores and subscribe to `GetPriceEvent` for anomalycore and calculate it there

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by VektorZ1** [comment]
It would be fun to add something like 5 damage (electricity for electro, heat for pyro, etc) in melee for inert core

**Comment by TheShuEd** [comment]
With the next PRs, I will add functionality to them. But different, not damage

File: Content.Server/Anomaly/Effects/AnomalyCoreSystem.cs (hunk 1 above)
Comment (TheShuEd) [reply Content.Server/Anomaly/Effects/AnomalyCoreSystem.cs]: Done

**Comment by TheShuEd** [comment]
```csharp
    public override void Update(float frameTime)
    {
        base.Update(frameTime);

        var query = EntityQueryEnumerator<AnomalyCoreComponent>();
        while (query.MoveNext(out var uid, out var component))
        {
            if (component.IsDecayed)
                continue;

            //When time runs out, we completely decompose
            if (component.DecayMoment < _gameTiming.CurTime)
                Decay(uid, component);
        }
    }
```
Is this an effective way to postpone code execution for a certain amount of time? I think there may be another, more correct method

**Comment by mirrorcult** [comment]
zumbased

**Comment by TheShuEd** [comment]
time to make a sequel to PR

**Pull request merged**